parachains/runtimes/assets/asset-hub-kusama/src/xcm_config.rs

```diff
@@ -503,3 +531,102 @@ impl pallet_assets::BenchmarkHelper<MultiLocation> for XcmBenchmarkHelper {
 		MultiLocation { parents: 1, interior: X1(Parachain(id)) }
 	}
 }
+
+/// All configuration related to bridging
+pub mod bridging {
+	use super::*;
+	use assets_common::{matching, matching::*};
+	use sp_std::collections::btree_set::BTreeSet;
+	use xcm_builder::{NetworkExportTable, UnpaidRemoteExporter};
+
+	parameter_types! {
+		pub BridgeHubKusamaParaId: u32 = 1002;
+		pub BridgeHubKusama: MultiLocation = MultiLocation::new(1, X1(Parachain(BridgeHubKusamaParaId::get())));
+		pub BridgeHubKusamaWithBridgeHubPolkadotInstance: MultiLocation = MultiLocation::new(1, X2(Parachain(BridgeHubKusamaParaId::get()), PalletInstance(53)));
+		pub const PolkadotNetwork: NetworkId = NetworkId::Polkadot;
+		pub AssetHubPolkadot: MultiLocation =  MultiLocation::new(2, X2(GlobalConsensus(PolkadotNetwork::get()), Parachain(1000)));
+		pub DotLocation: MultiLocation =  MultiLocation::new(2, X1(GlobalConsensus(PolkadotNetwork::get())));
+
+		/// Setup exporters configuration.
+		pub BridgeTable: sp_std::vec::Vec<(NetworkId, MultiLocation, Option<MultiAsset>)> = sp_std::vec![
+			(PolkadotNetwork::get(), BridgeHubKusama::get(), None)
+		];
+
+		/// Setup trusted bridged reserve locations.
+		/// Means that runtime accepts reserved assets from these locations.
+		pub TrustedBridgedReserveLocations: sp_std::vec::Vec<FilteredLocation> = sp_std::vec![
+			// trust assets from AssetHubPolkadot
+			(
+				AssetHubPolkadot::get(),
+				AssetFilter::ByMultiLocation(
+					MultiLocationFilter::default()
+						// allow receive DOT
+						.add_equals(DotLocation::get())
+						// and nothing else
+				)
+			)
+		];
+
+		/// Allowed reserve transfer assets per destination.
+		/// Means that runtime allows to transfer reserve assets to these locations.
+		pub AllowedReserveTransferAssetsLocations: sp_std::vec::Vec<FilteredLocation> = sp_std::vec![
+			// allow to transfer assets to AssetHubPolkadot
+			(
+				AssetHubPolkadot::get(),
+				AssetFilter::ByMultiLocation(
+					MultiLocationFilter::default()
+						// allow send only KSM
+						.add_equals(KsmLocation::get())
+						// and nothing else
+				)
+			)
+		];
```

Comment:
I don't quite understand this. Do you mean that AHK can only act as a reserve location for KSM to AHP? It should also act as a reserve location for all assets to other chains within Kusama (as per `type IsReserve`'s comment).

Reply:
yes, as I wrote in description:
- **explicitly is allowed only reserve based transfer for KSM and DOT and nothing else** at first (lets start defensively)
  - if we decided, we can allow whatever (ETH, TrustBackedAssets, ForeignAssets, PoolAssets ...)

and my mistake, I forgot to add **to cross-consensus AssetHubs**, so:
- **explicitly is allowed only reserve based transfer for KSM/DOT to cross-consensus AssetHubs and nothing else** at first (lets start defensively)
  - if we decided, we can allow whatever (ETH, TrustBackedAssets, ForeignAssets, PoolAssets ...)

---

so this `AllowedReserveTransferAssetsLocations` is fine-grained control if what we allow to transfer to remote AssetHub,
here:
```
        // Before:
        type XcmReserveTransferFilter = Everything

        // After:
        // Allow reserve based transfer to everywhere except for bridging, here we strictly check what assets are allowed.
	type XcmReserveTransferFilter =
		EverythingBut<bridging::IsNotAllowedExplicitlyForReserveTransfer>;
```


so in the PR, AssetHubKusama:
- act as a reserve location for all assets to other chains within Kusama  (like was here before with `Everything`)
- plus it acts as a reserve location for only KSM to AssetHubPolkadot

I just wanted to be pretty defensive, because I dont remember this was specified somewhere.

Or should we allow "ETH, TrustBackedAssets, ForeignAssets, PoolAssets, ..." to be transfered from AssetHubKusama to the AssetHubPolkadot?

Reply:
I also added tests for this `XcmReserveTransferFilter` with "almost all" possible combinations:

https://github.com/paritytech/cumulus/pull/2762/files#diff-eb0ad709a2e7c716cc9d0c86b1671e48f0cfb852b3bacd517ec1eab59e36cb29R713-R781

Reply:
> Or should we allow "ETH, TrustBackedAssets, ForeignAssets, PoolAssets, ..." to be transfered from AssetHubKusama to the AssetHubPolkadot?

No, I agree with the decision. I just wanted to make sure the comment expresses what it's actually doing. As long as it doesn't enter into this check unless the destination is over a bridge, then it's OK from the code side.